### PR TITLE
feat: integrate rabby wallet extension

### PR DESCRIPTION
**Pull request by RanGojo**
# Summary

integrate rabby wallet extension

Fixes # 1496

# How did you test this change?

Tested by installing Rabby wallet extension on Chrome browser and signing a transaction on BSC.

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Implemented a user interface (UI) change, referencing our Figma design to ensure pixel-perfect precision.

### CODE REVIEW

File: wallets/provider-rabby/src/index.ts
Comment (mikasackermn) [inline wallets/provider-rabby/src/index.ts:1]: Screenshot from 2024-06-08 12-38-30

I receive this error when installing both Rabby wallet and Metmasak.
Exists a solution for this?

File: wallets/provider-rabby/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+export const canEagerConnect: CanEagerConnect = canEagerlyConnectToEvm;
+
+export const getWalletInfo: (allBlockChains: BlockchainMeta[]) => WalletInfo = (
+  allBlockChains
+) => {
+  const evms = evmBlockchains(allBlockChains);
+  return {
+    name: 'Rabby',
+    img: 'https://raw.githubusercontent.com/rango-exchange/assets/main/wallets/metamask/icon.svg',
```
Comment (mikasackermn) [inline wallets/provider-rabby/src/index.ts:60]: Also, set the appropriate wallet logo.

**Review by mikasackermn — COMMENTED** [review]
Thanks, I just added some minor comments.

File: wallets/provider-rabby/src/index.ts
Comment (RyukTheCoder) [reply wallets/provider-rabby/src/index.ts:1]: You can flip between Metamask and Rabby from Rabby Wallet extension.

Comment (mikasackermn) [reply wallets/provider-rabby/src/index.ts:1]: I appreciate your explanation.

File: wallets/provider-rabby/src/index.ts (hunk 1 above)
Comment (RyukTheCoder) [reply wallets/provider-rabby/src/index.ts]: Fixed

**Review by mikasackermn — COMMENTED** [review]
LGTM

**Review by mikasackermn — APPROVED** [review]

**Review by nikaaru — APPROVED** [review]
thanks
LGTM

File: wallets/provider-rabby/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+export const canEagerConnect: CanEagerConnect = canEagerlyConnectToEvm;
+
+export const getWalletInfo: (allBlockChains: BlockchainMeta[]) => WalletInfo = (
+  allBlockChains
+) => {
+  const evms = evmBlockchains(allBlockChains);
+  return {
+    name: 'Rabby',
+    img: 'https://raw.githubusercontent.com/rango-exchange/assets/main/wallets/rabby/rabby.svg',
```
Comment (RanGojo) [inline wallets/provider-rabby/src/index.ts:60]: Please change the link here. @RyukTheCoder 
```suggestion
    img: 'https://raw.githubusercontent.com/rango-exchange/assets/main/wallets/rabby/icon.svg',
```

Comment (RyukTheCoder) [reply wallets/provider-rabby/src/index.ts]: Done

**Pull request merged**